### DISCUSSION TITLE
ImGui package not working

### DISCUSSION THREAD

**Discussion by MechSlayer** [dotnet/Silk.NET#468] [Q&A]
Hi, I tried to use the ImGui package following the Lab/ImGui example, but for some reason is not showing anything.
I tried using OpenGL 4.6 Core with a GLFW window.

Here is an example
```cs
//ctor
// window.View is a IWindow, and window.Input is a IInputContext
GuiController = new ImGuiController(glContext, window.View, window.Input);

//OnRender
GuiController.Update((float) deltaTime);
GlContext.Clear((uint) GLEnum.ColorBufferBit);
ImGui.ShowDemoWindow();
GuiController.Render();
Window.SwapBuffers();
```

**Comment by Perksey** [comment]
Either remove the extra `SwapBuffers` call or disable `ShouldSwapAutomatically` in `WindowOptions`

**Reply by MechSlayer** [reply]
Neither worked, I get the same result if I set ShouldSwapAutomatically to false and call SwapBuffers(), or if I set it to true and don't call SwapBuffers. I just tried copy/pasting the code from Lab/ImGui and I get the same result

**Reply by Perksey** [reply]
What do you actually see on your window? A black screen? Something else?

**Reply by MechSlayer** [reply]
Just the ClearScreen color, I did some more testing, and if I call `context.GetError()`, after calling the ImGuiController's render method, there is an InvalidEnum error

**Reply by Perksey** [reply]
What driver are you using?

**Reply by MechSlayer** [reply]
Latest AMD Adrenalin, the GPU is an RX 580

**Reply by Perksey** [reply]
Try running RenderDoc to see if you can find out which function is failing, which would help us a lot.

Also, check the output of `gl.GetString(StringName.Version)`

**Reply by MechSlayer** [reply]
This is the output of GetString `4.6.14757 Core Profile Context 20.12.1 27.20.14501.28009`.
Looks like all `glDrawElementsBaseVertex` are failing, this is the description of the error
```
No vertex buffer bound to attribute 0: in_color (buffer slot 0) at draw!
```
same goes for attribute 1, and 2, and for in_position and in_texCoord.
I can send the capture if you want.

**Reply by Perksey** [reply]
Yes that’ll be helpful :)

**Comment by MechSlayer** [comment]
Okay, I think the NuGet package has something wrong, because compiling the ImGui extension project works, but using the NuGet one fails.

**Reply by Perksey** [reply]
It's worth noting we've recently rewritten the ImGui extension, so 2.4 will have a brand new implementation which should be compatible with your existing code and fix your issue.
